crates/accelerate/src/quantum_circuit/circuit_instruction.rs

```diff
@@ -177,63 +177,59 @@ impl CircuitInstruction {
     // like that via unpacking or similar.  That means that the `parameters` field is completely
     // absent, and the qubits and clbits must be converted to lists.
     pub fn _legacy_format(&self, py: Python<'_>) -> PyObject {
-        PyTuple::new(
+        PyTuple::new_bound(
             py,
             [
-                self.operation.as_ref(py),
-                self.qubits.as_ref(py).to_list(),
-                self.clbits.as_ref(py).to_list(),
+                self.operation.bind(py),
+                &self.qubits.bind(py).to_list(),
+                &self.clbits.bind(py).to_list(),
             ],
         )
         .into_py(py)
     }
 
-    pub fn __getitem__(&self, py: Python<'_>, key: &PyAny) -> PyResult<PyObject> {
-        Ok(self
-            ._legacy_format(py)
-            .as_ref(py)
-            .get_item(key)?
-            .into_py(py))
+    pub fn __getitem__(&self, py: Python<'_>, key: &Bound<PyAny>) -> PyResult<PyObject> {
+        Ok(self._legacy_format(py).bind(py).get_item(key)?.into_py(py))
     }
 
     pub fn __iter__(&self, py: Python<'_>) -> PyResult<PyObject> {
-        Ok(self._legacy_format(py).as_ref(py).iter()?.into_py(py))
+        Ok(self._legacy_format(py).bind(py).iter()?.into_py(py))
     }
```

Comment:
It likely doesn't matter much at all, but is it worth changing `_legacy_format` to return a `Bound` here, rather than detaching it from the GIL only to immediately reattach it?

Reply:
Good idea. I believe I did this when first introducing this code because I'd wanted to use the `get_item` from `PyAny`, since its `key` argument can be a `&PyAny` instead of a `usize` (which is required when calling `get_item` on a `PyTuple`).

But, we can just use `as_any` on the `Bound<PyTuple>` and everything works out nicely.

Done in 1c6132d20e3f92e75cd1ae05031d6cb799369e79.